packages/developer-console-spa/src/components/lighthouse/LinkProjectForm.tsx

```diff
@@ -0,0 +1,205 @@
+    }
+    return (
+        <Form onSubmit={ handleSubmit( linkProject ) }>
+            { !props.branchVariant &&
+                <FormGroup
+                    isRequired
+                    fieldId=""
+                    label="Project Name"
+                    helperTextInvalid="Project is mandatory field"
+                    helperTextInvalidIcon={ <ExclamationCircleIcon /> }
+                    validated={ errors.project ? "error" : "default" }
+                >
+                    <Controller
+                        name="project"
+                        control={ control }
+                        render={ ( { field } ) => {
+                            return <ContextSelector
+                                { ...field }
+                                className="lighthouse-projectlist--filled"
+                                toggleText={ props.selectedProject.name }
+                                onSearchInputChange={ onSearchInputChange }
+                                isOpen={ isListOpen }
+                                searchInputValue={ searchTerm }
+                                onToggle={ onToggle }
+                                onSelect={ projectSelected }
+                                onSearchButtonClick={ onSearchButtonClick }
+                                screenReaderLabel="Selected Project:"
+                                footer={
+                                    <ContextSelectorFooter>
+                                        <Button variant="link" isInline onClick={ () => props.setActiveTabKey( 0 ) }>
+                                            Create a project
+                                        </Button>
+                                    </ContextSelectorFooter>
+                                }
+                            >
+                                { filteredProjects.map( ( item: any, index ) => (
+                                    <ContextSelectorItem key={ index }>{ item.name }</ContextSelectorItem>
+                                ) ) }
+                            </ContextSelector>;
+                        } }
+                    />
+                </FormGroup>
+            }
+            { props.selectedProject.id !== '' && <FormGroup
+                isRequired
+                fieldId=""
+                label="Select a branch"
+                helperTextInvalid="Branch is mandatory field"
+                helperTextInvalidIcon={ <ExclamationCircleIcon /> }
+                validated={ errors.branch ? "error" : "default" }
+                helperText="Start typing in the field to find a branch or create one if it does not exists."
+            >
+                <Controller
+                    name="branch"
+                    control={ control }
+                    render={ ( { field } ) => (
+                        <Select
+                            { ...field }
+                            { ...register( "projectName", { required: true } ) }
```

Comment:
This will be clash between hook form register and controller. Remove this register as controller is already used.

The controller name branch will be given as form field name. If its meant as projectName, change controller name to branch

Reply:
Fixed